### ISSUE TITLE
cannot remove "Save" restriction from LiveCycle Designer form

### ISSUE THREAD

**Issue by japm48** [qpdf/qpdf#464]
I am trying to remove the restrictions from [this document](https://www.juntadeandalucia.es/agenciadeserviciossocialesydependencia/es/docdownload/sol_depen_homologada.pdf/download) (no password).

To no avail, I tried to decrypt, reencrypt with good permissions and decrypt again:

```bash
f=original_file_name
qpdf --decrypt -- ${f}.pdf  ${f}_1.pdf
qpdf --encrypt '' '' 128 --accessibility=y --extract=y --assemble=y --annotate=y --form=y --print=full --modify=all -- ${f}_1.pdf ${f}_2.pdf
qpdf --decrypt -- ${f}_2.pdf ${f}_3.pdf
```

In particular, I'm interested in removing the "cannot save form" restriction. Adobe Reader won't let me save the form data introduced in any of the generated files (nor in the original).

This is not a problem with other PDF forms like [this one](https://campustecnologicoalgeciras.es/wp-content/uploads/2017/07/OoPdfFormExample.pdf).

BTW:
```bash
$ qpdf --version
qpdf version 10.0.1
```

**Comment by jberkenbilt** [comment]
Unfortunately, the ability to save a form from Adobe reader is not directly related to encryption. Adobe has this thing where it has some other path through digital signature where it won't let you save from *Adobe Reader* unless the original author permitted it, and this is done separately from the usual encryption method. Any transformation made by qpdf will invalidate the digital signature. This is not fixable because qpdf doesn't and will never have the private key. This would be true of any other non-Adobe software as well. That said, you should be able to save the form with the full Adobe Acrobat product or with many other PDF viewers that are capable of saving forms.

**Comment by japm48** [comment]
So, is it not possible to remove the original PDF signature without also removing the forms functionality?, at least this kind of forms, apparently there are [two types](https://en.wikipedia.org/wiki/PDF#Interactive_elements), I'm guessing this is using the newest format, XFA.

By "private key" do you refer to Adobe's private key? (otherwise, I understand there would be a method to use a new key)

Thanks for your help.

**Comment by jberkenbilt** [comment]
Without actually seeing the file, it's hard for me to say anything for sure, but the problem I'm describing is, as far as I know, not confined to XFA. The private key is Adobe's private key, but even if it weren't, qpdf doesn't at this time implement any of the PDF PKI features, though it is on my roadmap (no timeframe). Ultimately what I'm saying is that the feature of certain forms being authorized to be filled in Adobe Reader by the creator in Adobe Acrobat is more of an Acrobat feature than a PDF feature.

On Linux, you will find packages like atril, evince, or okular generally able to fill in and save forms regardless of whether this restriction is in place. I have heard that foxit is also able to do this, but I have personally never used that particular product.

Bottom line: if you have a form that can be filled out specifically in Adobe reader before qpdf touches it but not after qpdf touches it, then you are most likely bumping into this known and probably unfixable issue. Your workarounds are to use qpdf after you have filled in the form or to use some application other than Adobe reader to fill in the form. This could potentially include the full Adobe Acrobat product, but there are plenty of free alternatives.

All that said, without actually seeing the PDF file and reproducing the problem myself, I can't be 100% certain that your problem is an instance of this known issue. It just sounds very much like it is based on your description.

**Comment by japm48** [comment]
Thanks again for your assistance.
As this cannot ever be solved, please mark as "wontfix" or "invalid" if you prefer.
